src/apm_client.rs

```diff
@@ -69,11 +73,21 @@ impl ApmClient {
                 }
             })
             .map(Arc::new);
+        let client = if allow_invalid_certs {
+            let builder = reqwest::ClientBuilder::new().danger_accept_invalid_certs(true);
+
+            #[cfg(feature = "native-tls")]
+            #[cfg_attr(docsrs, doc(cfg(feature = "native-tls")))]
+            buidler.danger_accept_invalid_hostnames(true);
 
+            builder.build().unwrap_or(Client::new())
```

Comment:
Creating a default `Client` on failure can be something unexpected for the user. Everything would _seem_ to work, yet the new flag will be ignored. It's better to explicitly fail.